### ISSUE TITLE
Allow navController.push to specify the next view as root or to not include a back button

### ISSUE THREAD

**Issue by cfjedimaster** [driftyco/ionic#7741]
#### Short description of the problem:
NavController.push options should support the ability to tell the next view to not include a back button, or consider the next view as root.

#### What behavior are you expecting?
This is an ER, so it works as documented. ;)

Basically - push's 3rd arg, options, includes some stuff and I think an option for setRoot:true (default to false) could be useful. That or showBack:true/false.

**Comment by pedroigp** [comment]
http://ionicframework.com/docs/v2/api/components/nav/NavController/#setRoot
Set the root for the current navigation stack
Using setroot you wont have back button

**Comment by cfjedimaster** [comment]
Right - I should have said I know there is a workaround, but it "feels" natural to me to do a push and simply use an option for this. (imo)

**Comment by jgw96** [comment]
Hello @cfjedimaster . We just finished discussing this proposal and we decided that we are not in favor of adding this. Basically, we feel like using `setRoot` to set a view as a root is more explicit than setting a root with `push`. Also, since were pretty late in the game ( very close to an rc release ) and we have the `push` api nailed down pretty well, we'd rather not add functionality too it at this point. Thanks for using Ionic! 

ps: We always love to get suggestions and ideas from the community so keep them coming!
